python/ray/data/dataset.py

```diff
@@ -393,15 +393,24 @@ def parse_filename(row: Dict[str, Any]) -> Dict[str, Any]:
         logical_plan = LogicalPlan(map_op, self.context)
         return Dataset(plan, logical_plan)
 
+    @Deprecated(message="Use set_name instead", warning=True)
     def _set_name(self, name: Optional[str]):
+        self.set_name(name)
+
+    def set_name(self, name: Optional[str]):
```

Comment:
Why optional? Should this be required?

Reply:
I think this allows unsetting the name. 
Not sure if there are users doing this. 
But I'd like to keep it for backward compatibility.

Reply:
Wait, what's the use-case for the API? 

Why don't we just extend the ingress APIs (read_xxx) to specify the names that needed.

This API by itself looks quite dubious to be public API especially given the fact that it's not even chainable

Reply:
One use case is for training, where you can specify "train" and "validation" datasets.
This is the dataset-level name. So I don't think `read_xxx` is a proper name here.